packages/amplify-category-analytics/src/utils/pinpoint-helper.ts

```diff
@@ -84,3 +84,43 @@ export const pinpointHasInAppMessagingPolicy = (context: $TSContext): boolean =>
   const { cfnTemplate } = readCFNTemplate(pinpointCloudFormationTemplatePath, { throwIfNotExist: false }) || {};
   return !!cfnTemplate?.Parameters?.[pinpointInAppMessagingPolicyName];
 };
+
+/**
+ * checks if notifications category has a pinpoint resource - legacy projects
+ */
+export const checkIfNotificationsCategoryHasPinpoint = (context: $TSContext): $TSAny => {
+  const { amplify } = context;
+  const { amplifyMeta } = amplify.getProjectDetails();
```

Comment:
Assuming stateManager is imported, this can simplify to:

```suggestion
  const amplifyMeta = stateManager.getMeta();
```